src/geom_handlers/raster_processing.jl

```diff
@@ -0,0 +1,509 @@
+    # Mask out areas outside considered distance from port
+    return mask(Raster(target_rast; missingval=0); with=dist_buffer)
+end
+
+"""
+    calc_distances(
+        target_rast::Raster,
+        gdf::DataFrame;
+        units::String="NM"
+    )::Raster
+
+Calculate the minimum distance from each point in gdf.geometry to each valid pixel in target_rast.
+
+`target_rast` and the `gdf` should be in the same CRS (e.g., EPSG:7844 / GDA2020).
+
+# Arguments
+- `target_rast` : Raster of suitable pixels (Bool) to calculate distances from.
+- `gdf` : GeoDataFrame of 'points' for distance calculation.
+
+# Returns
+- Raster of distances from each cell to the closest point in gdf.
+"""
+function calc_distances(
+    target_rast::Raster,
+    gdf::DataFrame;
+    units::String="NM"
+)::Raster
+    tmp_areas = Float32.(copy(target_rast))
+
+    # First dimension is the rows (longitude)
+    # Second dimension is the cols (latitude)
+    raster_lon = Vector{Float64}(tmp_areas.dims[1].val)
+    raster_lat = Vector{Float64}(tmp_areas.dims[2].val)
+
+    @floop for (lon_ind, lon) in enumerate(raster_lon)
+        for (lat_ind, lat) in enumerate(raster_lat)
+            if tmp_areas[lon_ind, lat_ind] != 0.0
+                point = AG.createpoint()
+                AG.addpoint!(point, lon, lat)
+
+                pixel_dists = AG.distance.([point], gdf.geometry)
+                geom_point = gdf[argmin(pixel_dists), :geometry]
+                geom_point = (AG.getx(geom_point, 0), AG.gety(geom_point, 0))
+
+                dist_nearest = Distances.haversine(geom_point, (lon, lat))
+
+                # Convert from meters to nautical miles
+                if units == "NM"
+                    dist_nearest = dist_nearest / 1852
+                end
+
+                # Convert from meters to kilometers
+                if units == "km"
+                    dist_nearest = dist_nearest / 1000
+                end
+
+                tmp_areas.data[lon_ind, lat_ind] = Float32(dist_nearest)
+            end
+        end
+    end
+
+    tmp_areas = rebuild(tmp_areas, missingval=Float32(0.0))
+    return tmp_areas
+end
+
+"""
+    process_bottom_UTM_raster(
+        src_file::String,
+        dst_file::String,
+        target_crs::GFT.CoordinateReferenceSystemFormat,
+        target_missingval::Float64,
+        reg::String
+    )::Nothing
+
+Process bathymetry, slope and rugosity datasets from raw input data files and output to `output_fn`
+location. These datasets are in region UTM zone CRS.
+
+# Arguments
+- `src_file` : Path to raw bathymetry, slope or rugosity raster files in UTM CRS for processing.
+- `dst_file` : File location name to create output file. Should include variable and region information.
+- `target_crs` : Target CRS object to use in Rasters.resample(). e.g. using GFT.EPSG() format.
+- `target_missingval` : Consistent missingval to use in output raster.
+- `reg` : Region name for input CRS definition.
+
+"""
+function process_bottom_UTM_raster(
```

Comment:
Why "bottom"?

Reply:
I named them bottom just because the method should be applied to bathy, slope and rugosity layers, but I didn't want it to be assumed it could be applied to the wave layers which were also supposed to be in UTM zones but were incorrect. I can think of a better name here